src/gpu/generic/sycl/ref_inner_product.cpp

```diff
@@ -16,40 +16,810 @@
+            "destination memory descriptor is not a plain memory format");
+
+    // if anything contains a zero dimension, return success as this will be converted
+    // to a no-op
+
+    if (src_wrapper.has_zero_dim() || wei_wrapper.has_zero_dim()
+            || dst_wrapper.has_zero_dim()) {
+        has_zero_dim = true;
+        return status::success;
+    }
+
+    memory_desc_t src_reshaped;
+    memory_desc_t weights_reshaped;
+    memory_desc_t dst_reshaped;
+    format_tag_t src_format_tag = format_tag::ab;
+    format_tag_t wei_format_tag = format_tag::ba;
+    format_tag_t dst_format_tag = format_tag::ab;
+    memory_desc_t bias_reshaped = types::zero_md();
+
+    auto src_strides = src_wrapper.strides();
+    auto ndims = src_wrapper.ndims();
+
+    // If it's two dimensional, and not properly ordered, rely on TT/TN/NT GEMMs rather than reorders
+    if (ndims == 2) {
+        if (src_strides[1] != 1) { src_format_tag = format_tag::ba; }
+
+        if (wei_wrapper.strides()[1] != 1) { wei_format_tag = format_tag::ab; }
+    }
+
+    if (dst_wrapper.strides()[1] != 1) { dst_format_tag = format_tag::ba; }
+
+    dims_t src_squished_dims;
+    dims_t wei_squished_dims;
+    detail::get_flattened_dimension(
+            src_wrapper.dims(), src_squished_dims, src_wrapper.ndims());
+    detail::get_flattened_dimension(
+            wei_wrapper.dims(), wei_squished_dims, wei_wrapper.ndims(), true);
+    CHECK(memory_desc_init_by_tag(src_reshaped, 2, src_squished_dims,
+            src_wrapper.data_type(), src_format_tag));
+    CHECK(memory_desc_init_by_tag(weights_reshaped, 2, wei_squished_dims,
+            wei_wrapper.data_type(), wei_format_tag));
+    if (with_bias()) {
+        const auto bias_md = arg_md(DNNL_ARG_BIAS);
+        //Reshape bias to 1 x OC;
+        dims_t reshaped_bias_dims {1, bias_md->dims[0]};
+        CHECK(memory_desc_init_by_tag(bias_reshaped, 2, reshaped_bias_dims,
+                bias_md->data_type, format_tag::ab));
+    }
+
+    CHECK(memory_desc_init_by_tag(dst_reshaped, 2, dst_wrapper.dims(),
+            dst_wrapper.data_type(), dst_format_tag));
+
+    CHECK(gpu::generic::sycl::detail::init_matmul_pd(engine, attr(),
+            &src_reshaped, &weights_reshaped, &bias_reshaped, &dst_reshaped,
+            matmul_pd));
+
+    // check the memory format. (If 1 is not the innermost stride(ab...) or the second stride (a...b))
+    // reorder it to a ab... format. Only check src and weights, as dst will be handled by the matmul.
+    memory_desc_t src_reordered;
+
+    // Check if src needs reorde
+    bool is_favourable_layout
+            = (src_strides[ndims - 1] == 1 || src_strides[1] == 1)
+            && src_strides[0] == src_reshaped.dims[1];
+    bool is_not_strided = ((src_strides[0] == 1 || src_strides[1] == 1)
+            && src_wrapper.ndims() == 2); // check for strided plain layouts
+    is_favourable_layout = is_favourable_layout & is_not_strided;
```

Comment:
It's not a bitwise logic or operation.
```suggestion
    is_favourable_layout = is_favourable_layout && is_not_strided;
```

Reply:
Done